ext/curl/interface.c

```diff
@@ -232,6 +232,7 @@ static inline int build_mime_structure_from_hash(php_curl *ch, zval *zpostfields
 /* {{{ PHP_INI_BEGIN */
 PHP_INI_BEGIN()
 	PHP_INI_ENTRY("curl.cainfo", "", PHP_INI_SYSTEM, NULL)
+	PHP_INI_ENTRY("curl.user_agent", "", PHP_INI_ALL, NULL)
```

Comment:
After thinking about it more, I'd personally be in favor of PHP_INI_SYSTEM but opposed to the permissive PHP_INI_ALL. PHP_INI_SYSTEM would forbid calling ini_set at runtime, but still allow system configuration.

See https://www.php.net/manual/en/configuration.changes.modes.php and https://www.php.net/manual/en/configuration.changes.php (e.g. for web servers that use apache, the setting can be set in php.ini, and/or overridden for individual apache virtual hosts in /etc/httpd/vhosts.d/ with `php_value`) 

`php -d curl.user_agent='Some user agent' myCLIScript.php` would also always work if different CLI applications needed different user agents

-------

The reason I think the ability to call ini_set at runtime might be a problem:
1. If an application has two libraries that use the default curl user agent (libA and libB), and LibA were to call `ini_set('curl.user_agent', 'LibA/1.2.3')` and not bother to restore the user agent (or throw, etc), then the application would unexpectedly and spuriously sometimes call the endpoints that libB calls with the wrong user agent of LibA.

   Async frameworks using generators and php 8.1's fibers would make that a larger concern
2. So the application using LibB calling ServerB with a user agent from LibA may cause issues, such as developers wasting time debugging because of being mislead about what the service's caller actually is, or ServerB parsing the wrong versions from the user agent, or ServerB rejecting requests because the user agent is unrecognized or forbidden



Obviously, for `file_get_contents`, there's no way to override the user agent, so it's useful that `user_agent` can be overridden at runtime. But curl_setopt can already override the user agent

Reply:
Sorry for the late response. As far as i have checked you can override the user_agent during runtime using file_get_contents. Therefore i wanted to keep curl.user_agent in the same mode as the original user_agent.

This is my code to verify this:
```
<?php

ini_set('user_agent', 'test');

$context = stream_context_create([
  'http' => [
    'user_agent' => 'hello'
  ]
]);

$body = file_get_contents("https://www.whatsmyua.info/api/v1/ua", false, $conte$
$json = json_decode($body, true);

echo $json[0]["ua"]["rawUa"];
```

Reply:
> Sorry for the late response. As far as i have checked you can override the user_agent during runtime using file_get_contents. Therefore i wanted to keep curl.user_agent in the same mode as the original user_agent.

That only affects the call in question - I'm talking about ini_set affecting completely unrelated libraries that don't share a context

Reply:
That is correct, but wouldn't that also apply for the global user_agent option? If i have understood correctly the user_agent option will have the same "unwanted" affect. If we want to fix this problem, shouldn't we also limit the global user_agent option to PHP_INI_SYSTEM? (Which will be hard due to a quite hard BC break)

Anyway. I can gladly change to PHP_INI_SYSTEM if no one is against this change.

Reply:
> > Obviously, for file_get_contents, there's no way to override the user agent, so it's useful that user_agent can be overridden at runtime. But curl_setopt can already override the user agent
>
> Sorry for the late response. As far as i have checked you can override the user_agent during runtime using file_get_contents. Therefore i wanted to keep curl.user_agent in the same mode as the original user_agent.

Didn't realize that - I think the vast majority of calls to file_get_contents don't typically use a context, and curl is typically preferred (easier to set timeouts(lookup/request/connect), redirect behavior, allowed protocols, etc).

Also, in general, php aims for consistency within a module(aka php extension) (e.g. `ext/curl`) (individual modules do/may differ greatly from each other for naming, class/function usage, etc), which is why I personally strongly prefer PHP_INI_SYSTEM